vulntotal/datasources/osv.py

```diff
@@ -0,0 +1,169 @@
+# Copyright (c) nexB Inc. and others. All rights reserved.
+# http://nexb.com and https://github.com/nexB/vulnerablecode/
+# The VulnTotal software is licensed under the Apache License version 2.0.
+# Data generated with VulnTotal require an acknowledgment.
+#
+# You may not use this software except in compliance with the License.
+# You may obtain a copy of the License at: http://apache.org/licenses/LICENSE-2.0
+# Unless required by applicable law or agreed to in writing, software distributed
+# under the License is distributed on an "AS IS" BASIS, WITHOUT WARRANTIES OR
+# CONDITIONS OF ANY KIND, either express or implied. See the License for the
+# specific language governing permissions and limitations under the License.
+#
+# When you publish or redistribute any data created with VulnTotal or any VulnTotal
+# derivative work, you must accompany this data with the following acknowledgment:
+#
+#  Generated with VulnTotal and provided on an "AS IS" BASIS, WITHOUT WARRANTIES
+#  OR CONDITIONS OF ANY KIND, either express or implied. No content created from
+#  VulnTotal should be considered or used as legal advice. Consult an Attorney
+#  for any legal advice.
+#  VulnTotal is a free software tool from nexB Inc. and others.
+#  Visit https://github.com/nexB/vulnerablecode/ for support and download.
+
+import logging
+from typing import Iterable
+from urllib.parse import urljoin
+
+import requests
+from packageurl import PackageURL
+
+from vulntotal.validator import DataSource
+from vulntotal.validator import VendorData
+
+logger = logging.getLogger(__name__)
+
+
+class OSVDataSource(DataSource):
+    spdx_license_expression = "Apache-2.0"
+    license_url = "https://github.com/google/osv/blob/master/LICENSE"
+    url = "https://api.osv.dev/v1/query"
+
+    def fetch_advisory(self, payload):
+        response = requests.post(self.url, data=str(payload))
+        if not response.status_code == 200:
+            logger.error(f"Error while fetching {payload}: {response.status_code}")
+            return
+        return response.json()
+
+    def datasource_advisory(self, purl) -> Iterable[VendorData]:
+        payload = generate_payload(purl)
+        if not payload:
+            return
+        advisory = self.fetch_advisory(payload)
+        self._raw_dump.append(advisory)
+        return parse_advisory(advisory)
+
+    @classmethod
+    def supported_ecosystem(cls):
+        # source https://ossf.github.io/osv-schema/
+        return {
+        "npm": "npm", 
+        "maven": "Maven", 
+        "golang": "Go", 
+        "nuget": "NuGet", 
+        "pypi": "PyPI", 
+        "rubygems": "RubyGems", 
+        "crates.io": "crates.io", 
+        "composer": "Packagist", 
+        "linux": "Linux", 
+        "oss-fuzz": "OSS-Fuzz", 
+        "debian": "Debian", 
+        "hex": "Hex", 
+        "android": "Android", 
+    }
+
+
+def parse_advisory(response) -> Iterable[VendorData]:
+    if "vulns" in response:
+        for vuln in response["vulns"]:
```

Comment:
You can use a simpler approach and remove the line above `if "vulns" in response:` with this:
```suggestion
        for vuln in response.get("vulns") or []:
```